bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerEntry.java

```diff
@@ -53,22 +54,49 @@ public long getLength() {
         return length;
     }
 
+    /**
+     * Returns the content of the entry.
+     * This method can be called only once.
+     * While using v2 wire protocol this method will automatically release the internal ByteBuf
+     * @return
+     */
     public byte[] getEntry() {
+        if (data == null) {
```

Comment:
@sijie the first time getEntry is called we will set data to null, this way the client cannot re-use the buffer anymore.
On an eventual "second call" we will throw an IllegalStateException which explains the error.
if I drop the 'if' the user will get a NullPointerException, but this is not very clear and IMHO it is bad practice.

Unfortunately the ByteBuf can be used only once, because we have no clue to reset the original start position for the entry